### ISSUE TITLE
Assign `torch.Generator` in APIs like `torch.randn_like()`

### ISSUE THREAD

**Issue by KarhouTam** [pytorch/pytorch#128786]
### 🚀 The feature, motivation and pitch

It is possible to add `generator` argument in APIs like `torch.randn_like()`?
Because `torch.randn()` has argument `generator` and at the same time, `torch.randn_like(A, generator=rng_torch)` should output the same result as `torch.randn(A.shape, generator=rng_torch)`.

**Comment by albanD** [comment]
Yes this is an oversight on our end. We would accept a PR adding this argument to the random functions that don't have it yet.
